% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\title{625FinalReport}
\author{hrtang}
\date{2023-12-14}

\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={625FinalReport},
  pdfauthor={hrtang},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\begin{document}
\maketitle

Proteins, fundamental to biological systems, perform a diverse range of
functions vital for cellular and organism life, such as catalyzing
reactions, transferring signals, and immune system
(\textbf{alberts2022molecular?}). Accurately predicting these functions
is a significant challenge in biology, bridging the gap between an
ever-expanding database of protein sequences and their known functions.
Traditional experimental approaches, while effective, are often slow and
expensive. In contrast, computational methods, particularly deep
learning, offer a promising, cost-effective alternative due to their
ability to handle complex datasets and the availability of extensive
labeled data.

Deep Neural Networks (DNNs), a subset of Artificial Neural Networks
(ANNs), are particularly suited for protein function prediction. They
excel in extracting intricate features from basic input data, a crucial
aspect when dealing with the vast and complex data of protein sequences.
Our study focuses on leveraging these capabilities, using amino acid
sequences, without the 3-D structure, as inputs for DNN models. We chose
Gene Ontology (GO) terms as labels for protein functions, encompassing
three primary domains: molecular function, biological process, and
cellular component.

This research is part of a broader effort initiated by the Critical
Assessment of Functional Annotation (CAFA) competition, aimed at
advancing our understanding of protein functions. By developing models
trained on amino acid sequences, we aim to contribute to a deeper
understanding of how proteins determine cellular and organ function.
This insight is not only fundamental to biology but also has potential
applications in drug development and disease therapy.

\newpage

\hypertarget{references}{%
\paragraph{References}\label{references}}

\end{document}
